src/Data/Text/Lazy.hs

```diff
@@ -1427,18 +1429,32 @@ groupBy eq (Chunk t ts) = cons x ys : groupBy eq zs
 -- | /O(n)/ Return all initial segments of the given 'Text',
 -- shortest first.
 inits :: Text -> [Text]
-inits = (Empty :) . inits'
+inits = (NE.toList P.$!) . initsNE
+
+-- | /O(n)/ Return all initial segments of the given 'Text',
```

Comment:
The $O(n)$ claim is incorrect, this is $O(n^2)$ due to `L.map (Chunk t) (inits' ts)`.
But this was not introduced in the PR, so maybe it can be fixed separately.

Reply:
Alright, I changed the documentation for `Data.Text.Lazy.inits` and `Data.Text.Lazy.initsNE` to say _O(n²)_.

Reply:
I do not follow your reasoning however.

* `map` will take time linear in the amount of chunks.
* `Chunk t` will take constant time.

So, it seems to me `map (Chunk t)` should be linear. What am I missing?

Reply:
`map (Chunk t)` is linear, and you do it once per chunk. That makes the complexity quadratic.

More accurately, the complexity is `O(cn)` where `c` is the number of chunks (and `n = Lazy.Text.length input`).